src/components/v5/common/ActionSidebar/partials/MultiSigSidebar/partials/MultiSigWidget/partials/ApprovalStep/ApprovalStep.tsx

```diff
@@ -0,0 +1,333 @@
+import clsx from 'clsx';
+import React, { useState, type FC, useEffect } from 'react';
+
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import {
+  type ColonyActionType,
+  type ColonyMultiSigFragment,
+  MultiSigVote,
+  type MultiSigUserSignatureFragment,
+} from '~gql';
+import { useEligibleSignees } from '~hooks/multiSig/useEligibleSignees.ts';
+import useAsyncFunction from '~hooks/useAsyncFunction.ts';
+import { ActionTypes } from '~redux';
+import Tooltip from '~shared/Extensions/Tooltip/Tooltip.tsx';
+import { notMaybe } from '~utils/arrays/index.ts';
+import { formatText } from '~utils/intl.ts';
+import { getMultiSigRequiredRole } from '~utils/multiSig.ts';
+import Button from '~v5/shared/Button/Button.tsx';
+import MenuWithStatusText from '~v5/shared/MenuWithStatusText/MenuWithStatusText.tsx';
+import ProgressBar from '~v5/shared/ProgressBar/ProgressBar.tsx';
+import { StatusTypes } from '~v5/shared/StatusText/consts.ts';
+import StatusText from '~v5/shared/StatusText/StatusText.tsx';
+
+import RemoveVoteButton from '../../../RemoveVoteButton/RemoveVoteButton.tsx';
+import Signees from '../../../Signees/Signees.tsx';
+import VoteButton from '../../../VoteButton/VoteButton.tsx';
+import { hasWeekPassed } from '../../utils.ts';
+import MultiSigPills from '../MultiSigPills/MultiSigPills.tsx';
+
+interface ApprovalStepProps {
+  threshold: number;
+  actionType: ColonyActionType;
+  multiSigData: ColonyMultiSigFragment;
+  initiatorAddress: string;
+  createdAt: string;
+}
+
+const ApprovalStep: FC<ApprovalStepProps> = ({
+  threshold,
+  actionType,
+  multiSigData,
+  initiatorAddress,
+  createdAt,
+}) => {
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+  const [isButtonPending, setIsButtonPending] = useState(false);
+
+  const isOwner = user?.walletAddress === initiatorAddress;
+
+  const requiredRole = getMultiSigRequiredRole(actionType);
+  const { eligibleSignees } = useEligibleSignees({
+    domainId: multiSigData.multiSigDomainId,
+    requiredRole,
+  });
+  const voteOrder = {
+    [MultiSigVote.None]: 2,
+    [MultiSigVote.Approve]: 0,
+    [MultiSigVote.Reject]: 1,
+  };
+  const cancelMultiSigMotion = useAsyncFunction({
+    submit: ActionTypes.MULTISIG_CANCEL,
+    error: ActionTypes.MULTISIG_CANCEL_ERROR,
+    success: ActionTypes.MULTISIG_CANCEL_SUCCESS,
+  });
+  const handleCancelMultiSig = () => {
+    cancelMultiSigMotion({
+      colonyAddress: colony.colonyAddress,
+      motionId: multiSigData?.nativeMultiSigId,
```

Comment:
I think we can go with `multiSigData.nativeMultiSigId`, it shouldn't ever be undefined or null